connect/runtime/src/main/java/org/apache/kafka/connect/runtime/health/ConnectClusterStateImpl.java

```diff
@@ -42,13 +45,25 @@ public ConnectClusterStateImpl(HerderProvider herderProvider) {
 
     @Override
     public Collection<String> connectors() {
+        final CountDownLatch connectorsLatch = new CountDownLatch(1);
         final Collection<String> connectors = new ArrayList<>();
         herderProvider.get().connectors(new Callback<java.util.Collection<String>>() {
             @Override
             public void onCompletion(Throwable error, Collection<String> result) {
                 connectors.addAll(result);
+                connectorsLatch.countDown();
```

Comment:
@C0urante would using a `FutureCallback` be more appropriate here? 
Also do you happen to know what happens if this request is not forwarded to the leader of the group? How much do we care here (not easy to say because this class is missing javadoc)?

You might want to check `completeOrForwardRequest` to see what we do on another call to `herder.connectors` method elsewhere.

Reply:
@kkonstantine thanks for pointing out the `FutureCallback` class, that's much better than the `CountDownLatch`-based approach.

The `ConnectClusterStateImpl` class doesn't have javadocs, but the `ConnectClusterState` interface does. There's no mention of expected behavior if the `ConnectClusterState.connectors()` method is invoked on or off the master node, but FWICT both the standalone and distributed herders rely on their local `ClusterConfigState` snapshots to provide the list of connectors for their own `connectors()` methods, so it doesn't appear forwarding to the leader will be necessary. Let me know if I'm missing anything here; I've tried to go over the code base as best as I can but it's always good to hear from someone more familiar with it.